tools/debian/convert_debian.py

```diff
@@ -154,59 +167,70 @@ def parse_security_tracker_file(advisories: dict[str, AdvisoryInfo], file_handle
+                                                        dsa_match.group(2))
 
-    # Enumerate advisories + version info from security-tracker.
-    with open(os.path.join(security_tracker_repo,
-                           SECURITY_TRACKER_DSA_PATH)) as handle:
-        parse_security_tracker_file(advisories, handle)
 
+def parse_webwml_files(advisories: Advisories, webwml_repo: str):
     file_path_map = {}
 
-    for root, dirs, files in os.walk(os.path.join(webwml_repo, WEBWML_SECURITY_PATH)):
+    for root, _, files in os.walk(
+            os.path.join(webwml_repo, WEBWML_SECURITY_PATH)):
         for file in files:
             file_path_map[file] = os.path.join(root, file)
 
     # Add descriptions to advisories from wml files
-    for key in advisories:
-        # TODO: use regex to remove suffix?
-        mapped_key_no_ext = key.lower().removesuffix('-1').removesuffix('-2').removesuffix('-3').removesuffix('-4')
-        val_wml = file_path_map.get(mapped_key_no_ext + ".wml")
-        val_data = file_path_map.get(mapped_key_no_ext + ".data")
+    for key, adv in advisories.items():
+
+        # remove potential extension (e.g. DSA-12345-2, -2 is the extension)
+        mapped_key_no_ext = (MATCH_EXTENSION_FROM_DSA.sub(key.lower(), ''))
+        val_wml = file_path_map.get(mapped_key_no_ext + '.wml')
+        val_data = file_path_map.get(mapped_key_no_ext + '.data')
 
         if not val_wml:
             print(mapped_key_no_ext)
         else:
-            with open(val_wml) as handle:
+            with open(val_wml, encoding='utf-8') as handle:
                 data = handle.read()
                 html = WML_DESCRIPTION_PATTERN.findall(data)[0]
                 res = markdownify.markdownify(html)
-                advisories[key].details = res
+                adv.details = res
 
         if not val_data:
             print(mapped_key_no_ext)
         else:
-            with open(val_data) as handle:
+            with open(val_data, encoding='utf-8') as handle:
                 data: str = handle.read()
                 report_date: str = WML_REPORT_DATE_PATTERN.findall(data)[0]
-                # Split by ',' here for the occasional case where there are two dates in the "publish" field
-                advisories[key].published = \
-                    datetime.datetime.strptime(report_date.split(",")[0], "%Y-%m-%d").isoformat() + "Z"
+                # Split by ',' here for the occasional case where there
+                # are two dates in the 'publish' field
+                adv.published = (datetime.datetime.strptime(
+                    report_date.split(',')[0], '%Y-%m-%d').isoformat() + 'Z')
+
+
+def write_output(output_dir: str, advisories: Advisories):
+    for key in advisories:
+        with open(os.path.join(output_dir, key + '.json'),
+                  'w',
+                  encoding='utf-8') as output_file:
+            output_file.write(str(advisories[key]))
```

Comment:
Let's be explicit here and call json.dumps() here on `advisory.to_dict()`. This makes it clear to the reader what's going on here.